c/parallel/src/merge_sort.cu

```diff
@@ -313,33 +320,41 @@ CUresult cccl_device_merge_sort_build(
 
     const std::string op_src = make_kernel_user_comparison_operator(input_keys_it_value_t, op);
 
+    constexpr std::string_view src_template = R"XXX(
+#include <cub/device/dispatch/kernels/merge_sort.cuh>
+#include <cub/util_type.cuh> // needed for cub::NullType
+struct __align__({1}) storage_t {{
+  char data[{0}];
+}};
+struct __align__({3}) items_storage_t {{
+  char data[{2}];
+}};
+{7}
+{8}
+{9}
+{10}
+struct agent_policy_t {{
+  static constexpr int ITEMS_PER_TILE = {6};
+  static constexpr int ITEMS_PER_THREAD = {5};
+  static constexpr int BLOCK_THREADS = {4};
+  static constexpr cub::BlockLoadAlgorithm LOAD_ALGORITHM = cub::BLOCK_LOAD_WARP_TRANSPOSE;
+  static constexpr cub::CacheLoadModifier LOAD_MODIFIER = cub::LOAD_LDG;
+  static constexpr cub::BlockStoreAlgorithm STORE_ALGORITHM = cub::BLOCK_STORE_WARP_TRANSPOSE;
+}};
+struct device_merge_sort_policy {{
+  struct ActivePolicy {{
+    using MergeSortPolicy = agent_policy_t;
+  }};
+}};
+struct device_merge_sort_vsmem_helper {{
+  template<typename ActivePolicyT, typename... Ts>
+  using MergeSortVSMemHelperT = cub::detail::merge_sort::VSMemHelper::MergeSortVSMemHelperT<ActivePolicyT, Ts...>;
+}};
```

Comment:
important: the idea behind this indirection level was that it'd allow us to disable vsmem helper on the kernel end. Right now, host side on C parallel does something along the lines of: `int BlockThreads(policy) { return policy.block_threads; }`, in other words, it doesn't modify the block size in any way. But the `cub::detail::merge_sort::VSMemHelper::MergeSortVSMemHelperT` can modify the block size based on the key / value type sizes. We need another virtual shared memory helper in this string that does something like:

```
struct vsmem_helper
{
  using policy_t = agent_policy_t;
  // ...
};
```